SLMM.Domain/LawnMap/RectLawnMap.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Drawing;
+
+namespace SLMM.Domain.LawnMap
+{
+    public class RectLawnMap : IRectLawnMap
+    {
+        public Rectangle Geometry { get; set; }
+
+        public RectLawnMap(Rectangle geometry)
+        {
+            Geometry = geometry;
+
+            if (Geometry.IsEmpty)
+                throw new ArgumentOutOfRangeException(nameof(geometry));
+        }
+
+        public bool CanMove(Point coordinates)
+        {
+            return Geometry.Contains(coordinates);
+        }
+
+        public bool CanMove(in Position position, int steps)

```

Comment:
`CanMove` mixes responsibilities of calculating position and finding out whether it is possible to move there.

Much better would be to have

```
bool CanMoveInto(Point location);
```

Reply:
See above comment regarding TryMove.

P.S CanMoveInto sounds way better and clearly communicates the intention, I like it 👍